### PR TITLE
[PLAT-1145][Hotfix] Confirming ORCID account doesn't work if you are logged in

### PR DESCRIPTION
**Pull request by Johnetordoff**
## Purpose

If you try to confirm your ORCID account while logged in the OSF will redirect to the dashboard without confirming the account. This fix allows you to comfirm it without logging out.

## Changes

- moves redirect clause to after confirmation clause so it's not missed.
- adds test

## QA Notes

I tested this in the shell by creating a user with (paraphasing) :
```python 
user.email_verifications = {'token': {'email': user.email, 'external_identity': {'ORCID' {'fake_external_identity_id': 'fake status'}}}}
me.external_identity = {'ORCID': {'fake_external_identity_id': 'fake status'}}
```
and hitting endpoint `http://localhost:5000/confirm/external/<user guid>/token/?destination=dashboard` then checking account setting to see if the user had a confirmed fake ORCID.

## Documentation

🐞 fix, no docs

## Side Effects

None that I know of.

## Ticket

https://openscience.atlassian.net/browse/PLAT-1145

### CODE REVIEW

File: tests/test_views.py [hunk 1]
```diff
@@ -3942,6 +3942,23 @@ def test_external_login_confirm_email_get_create(self, mock_welcome):
         assert_true(self.user.is_registered)
         assert_true(self.user.has_usable_password())
 
+    def test_external_login_confirm_email_get_while_logged_in(self):
+        external_identity = {
+            'orcid': {
+                self.provider_id: 'PENDING'
+            }
+        }
+        user = AuthUserFactory(external_identity=external_identity)
```
Comment (erinspace) [inline tests/test_views.py]: Could you modify `self.user` for your test instead of taking the time to create your own?

File: framework/auth/views.py [hunk 2]
```diff
@@ -543,6 +526,23 @@ def external_login_confirm_email_get(auth, uid, token):
     user.verification_key = generate_verification_key()
     user.save()
 
+    # if user is already logged in
```
Comment (erinspace) [inline framework/auth/views.py]: Curious of how much of the above should indeed be happening before the current logged in status of the user is checked - for example `user.date_last_logged_in = timezone.now()` should probably still happen after the logged in check.

File: framework/auth/views.py [hunk 3]
```diff
@@ -564,6 +546,25 @@ def external_login_confirm_email_get(auth, uid, token):
...
+            # external domain takes priority
+            campaign_url = campaigns.external_campaign_url_for(destination)
+            if not campaign_url:
+                campaign_url = campaigns.campaign_url_for(destination)
+            return redirect(campaign_url)
+        if new:
+            status.push_status_message(language.WELCOME_MESSAGE, kind='default', jumbotron=True, trust=True, id='welcome_message')
+        return redirect(web_url_for('dashboard'))
+
+    user.date_last_logged_in = timezone.now()
```
Comment (sloria) [inline framework/auth/views.py]: Why was this moved? It looks like this change will never get saved.

Comment (Johnetordoff) [reply framework/auth/views.py]: I'm not sure what you mean, the idea with this is that if the user was logged in already their `date_last_logged_in` shouldn't change, but if they log in with this function it should miss the redirect and hit that line.

Comment (sloria) [reply framework/auth/views.py]: OK, but the user isn't being saved after this point.
